code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -0,0 +1,207 @@
+    private static final String INSPECT_CONFIG = "Config";
+    private static final String INSPECT_IMAGE = "Image";
+    private static final String IMAGE_PREFIX = "docker:";
+    private static final String IMAGE_KIND_PATTERN = "^(sim|docker):.*";
+    private static final String RANCHER_UUID = "RANCHER_UUID=";
+    private static final String RANCHER_NETWORK = "RANCHER_NETWORK=";
+
+    @Inject
+    StorageService storageService;
+
+    @Inject
+    NetworkDao networkDao;
+
+    @Inject
+    AccountDao accountDao;
+
+    @Inject
+    LockManager lockManager;
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public HandlerResult handle(ProcessState state, ProcessInstance process) {
+        if ( !MANAGE_NONRANCHER_CONTAIERS.get() ) {
+            return null;
+        }
+
+        ContainerEvent event = (ContainerEvent)state.getResource();
+        Map<String, Object> inspect = (Map<String, Object>)DataUtils.getFields(event).get(FIELD_DOCKER_INSPECT);
+
+        String rancherUuid = getRancherUuidLabel(inspect);
+        if ( rancherUuid != null ) {
+            return null;
+        }
+
+        scheduleInstance(event, inspect);
+
+        return null;
+    }
+
+    void scheduleInstance(final ContainerEvent event, final Map<String, Object> inspect) {
+        final Long accountId = event.getAccountId();
+        final String externalId = event.getExternalId();
+
+        lockManager.lock(new ContainerEventInstanceLock(accountId, externalId), new LockCallbackNoReturn() {
+            @Override
+            public void doWithLockNoResult() {
+                // TODO INDEX ON EXTERNAL ID
+                Instance instance = objectManager.findAny(Instance.class, INSTANCE.ACCOUNT_ID, accountId,
+                        INSTANCE.EXTERNAL_ID, externalId);
+                if ( instance == null ) {
+                    instance = objectManager.newRecord(Instance.class);
+                    instance.setKind(InstanceConstants.KIND_CONTAINER);
+                    instance.setAccountId(accountId);
+                    instance.setExternalId(externalId);
+                    setName(inspect, instance);
+                    setNetwork(accountId, inspect, instance);
+                    setImage(inspect, instance);
+                    setHost(event, instance);
+                    DataAccessor.fields(instance).withKey(InstanceConstants.FIELD_NATIVE_CONTAINER).set(Boolean.TRUE);
+                    instance = objectManager.create(instance);
+                }
+
+                if ( !InstanceConstants.STATE_RUNNING.equals(ObjectUtils.getState(instance)) &&
+                        !InstanceConstants.STATE_STARTING.equals(ObjectUtils.getState(instance))) {
+                    try {
+                        getObjectProcessManager().scheduleProcessInstance(InstanceConstants.PROCESS_CREATE, instance, makeData());
+                    } catch (ProcessCancelException e) {
+                        // TODO Figure out if i need this catch
```

Comment:
I usually put try/catch ProcessCancelException when I schedule the process for the object w/o verifying its original state